src/processor/reconnection-graph.service.ts

```diff
@@ -491,22 +491,10 @@ export class ReconnectionGraphService {
         { eventPallet: 'utility', event: 'BatchCompleted' },
         providerKeys, batch).signAndSend();
       
-      if (!event) {
+      if (!event|| !this.blockchainService.api.events.utility.BatchCompleted.is(event)) {
```

Comment:
if matching event sent to extrnsic helper is not met it instead throws an error which is handled in catch